spynnaker/pyNN/models/spike_source/spike_source_poisson_vertex.py

```diff
@@ -813,78 +535,32 @@ def get_dtcm_usage_for_atoms():
-        :param ~pacman.model.routing_info.RoutingInfo routing_info:
-        :param ~pacman.model.graphs.machine.MachineGraph graph:
-        :param int first_machine_time_step:
-        """
-        # pylint: disable=too-many-arguments, arguments-differ
-
-        # reserve the neuron parameters data region
-        self._reserve_poisson_params_rates_region(placement, spec)
-
-        # write parameters
-        self._write_poisson_parameters(
-            spec=spec, graph=graph, placement=placement,
-            routing_info=routing_info,
-            vertex_slice=placement.vertex.vertex_slice,
-            machine_time_step=machine_time_step)
-
-        # write rates
-        self._write_poisson_rates(
-            spec, placement.vertex.vertex_slice, machine_time_step,
-            first_machine_time_step)
-
-        # end spec
-        spec.end_specification()
-
-    @inject_items({"first_machine_time_step": "FirstMachineTimeStep"})
-    @overrides(AbstractRewritesDataSpecification
-               .requires_memory_regions_to_be_reloaded,
-               additional_arguments={"first_machine_time_step"})
-    def requires_memory_regions_to_be_reloaded(self, first_machine_time_step):
-        # pylint: disable=arguments-differ
-        return (self.__change_requires_neuron_parameters_reload or
-                first_machine_time_step == 0)
-
-    @overrides(AbstractRewritesDataSpecification.mark_regions_reloaded)
-    def mark_regions_reloaded(self):
-        self.__change_requires_neuron_parameters_reload = False
+    def kiss_seed(self, vertex_slice):
+        if vertex_slice not in self.__kiss_seed:
+            self.__kiss_seed[vertex_slice] = create_mars_kiss_seeds(
+                self.__rng, self.__seed)
+        return self.__kiss_seed[vertex_slice]
 
-    @overrides(AbstractReadParametersBeforeSet.read_parameters_from_machine)
+    @overrides(
+        AbstractReadParametersBeforeSet.read_parameters_from_machine)
     def read_parameters_from_machine(
             self, transceiver, placement, vertex_slice):
 
         # locate SDRAM address where parameters are stored
-        poisson_params = \
-            helpful_functions.locate_memory_region_for_placement(
-                placement, _REGIONS.POISSON_PARAMS_REGION.value, transceiver)
+        poisson_params = placement.vertex.poisson_param_region_address(
+            placement, transceiver)
         seed_array = transceiver.read_memory(
             placement.x, placement.y, poisson_params + SEED_OFFSET_BYTES,
             SEED_SIZE_BYTES)
         self.__kiss_seed[vertex_slice] = struct.unpack_from("<4I", seed_array)
 
         # locate SDRAM address where the rates are stored
-        poisson_rate_region_sdram_address = \
-            helpful_functions.locate_memory_region_for_placement(
-                placement, _REGIONS.RATES_REGION.value, transceiver)
+        poisson_rate_region_sdram_address = (
+            placement.vertex.poisson_rate_region_address(
```

Comment:
This is ugly but will be resolved when this method moves to the machine vertex.

Reply:
true